src/DrCabal/Profile.hs

```diff
@@ -52,16 +54,18 @@ readFromFile file = eitherDecodeFileStrict' file >>= \case
         exitFailure
     Right entries -> pure entries
 
-createProfileChart :: Int -> [Entry] -> Text
-createProfileChart width l = case l of
+type Formatter = Word64 -> Word64 -> Int -> Map Text Phase -> Text
+
+createProfileChart :: Formatter -> Int -> [Entry] -> Text
+createProfileChart formatter width l = case l of
     [] -> unlines
         [ "No cabal build entries found. Have you already built dependency?"
         , "Try removing global cabal store cache and rerunning 'dr-cabal watch' again."
         ]
     entries ->
         let start = List.minimum $ map entryStart entries in
         let end   = List.maximum $ map entryStart entries in
-        formatChart start end width $ calculatePhases start $ groupEntries entries
+        formatter start end width $ calculatePhases start $ groupEntries entries
```

Comment:
This `Formatter` type alias won't work as the `Timeline` module will have a different type instead of `Phase` and it'll have a different implementation of `calculatePhases`. So, for now, I propose to just move this function entirely to the `Stacked` module (you can rename it to `createStackedChart`) and we'll deal with duplication avoidance later

Reply:
Done.